python/afdko/makeotf.py

```diff
@@ -1223,17 +1234,13 @@ def getOptions(makeOTFParams, args):
                 fontRevisionString = "5"
 
             # It is not another option. Must be a valid number
-            formatOK = 0
-            try:
-                fontRevision = eval(fontRevisionString)
-                if isinstance(fontRevision, int):
-                    formatOK = 1
-                elif isinstance(fontRevision, float):
-                    _, minor = fontRevisionString.split(".")
-                    if len(minor) == 3:
-                        formatOK = 1
-            except ValueError:
-                pass
+            formatOK = False
+            frs_parts = fontRevisionString.split(".")
+            formatOK = frs_parts[0].isnumeric()
```

Comment:
What happens if someone uses `+1.000` or `-1.000` for revision number?

Reply:
a '+' or "-" before the numbers won't match either of the RegExes (for getting existing value from .fea file or user input with `-rev` CLI option), so `makeotf` won't modify the .fea file. I don't know what `makeotfexe` would do with a value like that if it's in a .fea file, that should probably be tested.

Reply:
More on this: `makeotf -rev -1.000` doesn't even get to that part of the code (the CLI arg parser doesn't recognize '-1.000' as an option). `makeotf -rev +1.000` trips the `if not formatOK` test. So I don't think this is a concern for `makeotf`. Again, probably should check what `makeotfexe` does when a .fea file contains values like that, but that's a different issue/PR, I think.